src/serac/physics/tests/solid_nonlinear_solve.cpp

```diff
@@ -56,59 +213,151 @@ void functional_solid_test_nonlinear_buckle()
 
-  // set linear elastic material
   serac::solid_mechanics::NeoHookean material{density, bulkMod, shearMod};
-  // serac::solid_mechanics::StVenantKirchhoff material{density, bulkMod, shearMod};
   seracSolid->setMaterial(serac::DependsOn<>{}, material);
 
   // fix displacement on side surface
   seracSolid->setDisplacementBCs({2, 3, 4, 5}, [](const mfem::Vector&, mfem::Vector& u) { u = 0.0; });
+  // seracSolid->setDisplacementBCs({3}, [](const mfem::Vector&, mfem::Vector& u) { u = 0.0; });
 
   serac::Domain topSurface = serac::Domain::ofBoundaryElements(*meshPtr, serac::by_attr<DIM>(6));
-  seracSolid->setTraction([&](auto, auto n, auto) { return -loadMagnitude * n; }, topSurface);
-
+  //seracSolid->setTraction([&](auto, auto n, auto) { return -loadMagnitude * n; }, topSurface);
+  seracSolid->setPressure([&](auto, auto) { return loadMagnitude; }, topSurface);
   seracSolid->completeSetup();
   seracSolid->advanceTimestep(1.0);
 
-  const serac::FiniteElementState& displacement = seracSolid->state("displacement");
-  mfem::ParGridFunction            uGF(const_cast<mfem::ParFiniteElementSpace*>(&displacement.space()));
-  displacement.fillGridFunction(uGF);
-
-  mfem::VisItDataCollection visit_dc("nonlinearPlate", meshPtr);
-  visit_dc.RegisterField("u", &uGF);
-  visit_dc.Save();
+  seracSolid->outputStateToDisk("paraview_buckle_easy");
 }
 
-TEST(SolidMechanics, nonlinear_solve) { functional_solid_test_nonlinear_buckle(); }
+TEST(SolidMechanics, nonlinear_solve_buckle_easy) { functional_solid_test_nonlinear_buckle(5e-10); }
+//TEST(SolidMechanics, nonlinear_solve_buckle_medium) { functional_solid_test_nonlinear_buckle(4e-4); }
+//TEST(SolidMechanics, nonlinear_solve_buckle_hard) { functional_solid_test_nonlinear_buckle(3e-2); }
+//TEST(SolidMechanics, nonlinear_solve_euler) { functional_solid_test_euler(); }
+
+class InputParser
+{
+public:
+  InputParser(int& argc, char** argv){
+    for (int i=1; i < argc; ++i) {
+      this->tokens.push_back(std::string(argv[i]));
+    }
+  }
+  std::string getCmdOption(const std::string& option) const {
+    std::vector<std::string>::const_iterator itr;
+    itr =  std::find(this->tokens.begin(), this->tokens.end(), option);
+    if (itr != this->tokens.end() && ++itr != this->tokens.end()) {
+        return *itr;
+    }
+    static const std::string empty_string("");
+    return empty_string;
+  }
+  bool cmdOptionExists(const std::string& option) const {
+    return std::find(this->tokens.begin(), this->tokens.end(), option)
+            != this->tokens.end();
+  }
+private:
+  std::vector <std::string> tokens;
+};
 
 int main(int argc, char* argv[])
 {
-  ::testing::InitGoogleTest(&argc, argv);
-  MPI_Init(&argc, &argv);
+  InputParser parser(argc, argv);
+  auto filename = parser.getCmdOption("-p");
+  if (!filename.empty()) {
+    mesh_path = filename;
+  }
 
-  axom::slic::SimpleLogger logger;
+  axom::CLI::App app{"Nonlinear problems"};
```

Comment:
```suggestion
  axom::CLI::App app{"Nonlinear problems"};
  app.add_option("-p", mesh_path, "Mesh file to use")->check(axom::CLI::ExistingFile);
```

Sorry for the confusing and deleted comment, I tried to do a longer suggestion but Github won't let you do suggestions over any line that is deleted, which is confusing. 

This should remove the need for the `InputParser` class above. I didn't test it but it is copy/pasted from our existing CLI code.

Reply:
Fixed